app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageRecipientFieldsManager.kt

```diff
@@ -0,0 +1,184 @@
+/*
+ * Infomaniak Mail - Android
+ * Copyright (C) 2023 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.ui.newMessage
+
+import android.view.View
+import androidx.core.view.isGone
+import androidx.core.view.isVisible
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
+import com.infomaniak.lib.core.utils.showKeyboard
+import com.infomaniak.mail.data.models.correspondent.MergedContact
+import com.infomaniak.mail.data.models.correspondent.Recipient
+import com.infomaniak.mail.data.models.draft.Draft
+import com.infomaniak.mail.databinding.FragmentNewMessageBinding
+import com.infomaniak.mail.ui.newMessage.NewMessageRecipientFieldsManager.FieldType.*
+import com.infomaniak.mail.utils.MergedContactDictionary
+import com.infomaniak.mail.utils.copyRecipientEmailToClipboard
+import dagger.hilt.android.scopes.ActivityScoped
+import javax.inject.Inject
+
+@ActivityScoped
+class NewMessageRecipientFieldsManager @Inject constructor() : NewMessageManager() {
+
+    private var _externalsManager: NewMessageExternalsManager? = null
+    private val externalsManager: NewMessageExternalsManager get() = _externalsManager!!
```

Comment:
shouldn't this be obtain by injection ?

Reply:
If we do end up using it in more places than this single one, we could go this route but for now I prefer this way as it better shows that externalManager needs to init its own values before being accessed by editorManager. Same for the other manager